modules/samples/src/test/java/org/lwjgl/demo/openxr/HelloOpenXR.java

```diff
@@ -0,0 +1,835 @@
+            //Initialize OpenXR's OpenGL compatability
+            XrGraphicsRequirementsOpenGLKHR graphicsRequirements = XrGraphicsRequirementsOpenGLKHR.mallocStack();
+            graphicsRequirements.set(KHROpenglEnable.XR_TYPE_GRAPHICS_REQUIREMENTS_OPENGL_KHR, 0, 0, 0);
+            KHROpenglEnable.xrGetOpenGLGraphicsRequirementsKHR(xrInstance, systemID, graphicsRequirements);
+
+            //Init glfw
+            if (!glfwInit()) {
+                throw new IllegalStateException("Failed to initialize GLFW.");
+            }
+            glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 4);
+            glfwWindowHint(GLFW_CONTEXT_VERSION_MINOR, 0);
+            glfwWindowHint(GLFW_OPENGL_PROFILE, GLFW_OPENGL_CORE_PROFILE);
+            glfwWindowHint(GLFW_DOUBLEBUFFER, GL_FALSE);
+            window = glfwCreateWindow(640, 480, "Hello World", NULL, NULL);
+            glfwMakeContextCurrent(window);
+            GL.createCapabilities();
+
+            //Check if OpenGL ver is supported by OpenXR ver
+            if (graphicsRequirements.minApiVersionSupported() >= XR10.XR_MAKE_VERSION(GL11.glGetInteger(GL30.GL_MAJOR_VERSION), GL11.glGetInteger(GL30.GL_MINOR_VERSION), 0)) {
+//                throw new IllegalStateException("Runtime does not support desired Graphics API and/or version");]
+                System.err.println("Runtime does not support desired Graphics API and/or version");
+            }
+
+            //Bind the OpenGL context to the OpenXR instance and create the session
+            switch (Platform.get()) {
+                case LINUX:
+//                    if (xlib) { TODO
+//                        XrGraphicsBindingOpenGLXlibKHR graphicsBinding = XrGraphicsBindingOpenGLXlibKHR.malloc();
+//                        graphicsBinding.set(
+//                            KHROpenglEnable.XR_TYPE_GRAPHICS_BINDING_OPENGL_XLIB_KHR,
+//                            NULL,
+//                            GLFWNativeX11.glfwGetX11Display(),
+//                            ,
+//                            ,
+//                            GLX.glXGetCurrentDrawable(),
+//                            GLFWNativeGLX.glfwGetGLXContext(window)
+//                        );
+//                        this.graphicsBinding = graphicsBinding;
+//                    } else if (wayland) {
+//                        XrGraphicsBindingOpenGLWaylandKHR graphicsBinding = XrGraphicsBindingOpenGLWaylandKHR.malloc();
+//                        graphicsBinding.set(
+//                            KHROpenglEnable.XR_TYPE_GRAPHICS_BINDING_OPENGL_WAYLAND_KHR,
+//                            NULL,
+//                            GLFWNativeWayland.glfwGetWaylandDisplay()
+//                        );
+//                        this.graphicsBinding = graphicsBinding;
+//                    } else {
+//                        throw new IllegalStateException();
+//                    }
+                    throw new IllegalStateException();
```

Comment:
It would be nice if we could fix the linux support for the OpenGL example before finishing the pull request. Unfortunately, I have never really worked with windowing libraries directly, so I wouldn't know how to do this either. If I have time left, I will give it a try. Also, I think it would be better to create some static utility method that handles this platform-specific case distinction rather than letting users copy this code into all of their projects.

Reply:
I've done a bit of research into getting XLib support working for the OpenGL and I've come to the conclusion that there are two ways to get it to work. 
* Ditch GLFW entirely in the example (for XLib) and create the window and OpenGL context manually
* Update GLFW to expose some of the needed parameters

While the first solution will most likely work it feels much too hacky for LWJGL. However, the other will require updating both GLFW and the GLFW bindings so will take much longer to be approved despite being a much better long term solution.

Reply:
The second option sounds best to me. Why would that take much more time? Not many extra functions are needed, right? (At least, nothing compared to the number of functions this pull request will add for OpenXR.)

I dislike the first option because I would like to minimize platform-dependent code. Such code is nasty because testing it requires checking for each platform.